File: back-end/src/main/java/com/example/backend/dto/school/SchoolDTO.java
package com.example.backend.dto.school;

import com.example.backend.dto.base.BaseDTO;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SchoolDTO extends BaseDTO {
    private String name;
    private String address;
    private String phoneNumber;
    private Double latitude;
    private Double longitude;
    private Long activeStudentsCount;
    private Long activeRoutesCount;
    private Long activeDriversCount;
    private Long activeVehiclesCount;
} 

File: front-end/src/core/dto/school.dto.ts
import type { BaseDTO } from "./base.dto"

export interface SchoolDTO extends BaseDTO {
  name: string
  address: string
  phoneNumber: string
  latitude?: number
  longitude?: number
}

export interface CreateSchoolRequest {
  name: string
  address: string
  phoneNumber: string
  latitude?: number
  longitude?: number
  // adminId is handled by the backend based on the authenticated user
}

export interface UpdateSchoolRequest {
  name?: string
  address?: string
  phoneNumber?: string
  latitude?: number
  longitude?: number
}

